butane_core/src/db/turso.rs

```diff
@@ -0,0 +1,1230 @@
+    })
+}
+
+/// Convert a Turso value to SqlVal without type information.
+/// This infers the SqlVal type from the Turso value type.
+fn turso_value_to_sqlval_untyped(val: &turso::Value) -> Result<SqlVal> {
+    Ok(match val {
+        turso::Value::Null => SqlVal::Null,
+        turso::Value::Integer(i) => SqlVal::BigInt(*i),
+        turso::Value::Real(r) => SqlVal::Real(*r),
+        turso::Value::Text(t) => SqlVal::Text(t.clone()),
+        turso::Value::Blob(b) => SqlVal::Blob(b.clone()),
+    })
+}
+
+// Turso row wrapper that stores owned SqlVal values
+struct TursoRow {
+    values: Vec<SqlVal>,
+}
+
+impl BackendRow for TursoRow {
+    fn get(&self, idx: usize, ty: SqlType) -> Result<SqlValRef<'_>> {
+        self.values
+            .get(idx)
+            .ok_or_else(|| Error::Internal(format!("Column index {} out of bounds", idx)))
+            .and_then(|val| {
+                if val.is_compatible(&ty, true) {
+                    Ok(val)
+                } else {
+                    Err(Error::CannotConvertSqlVal(ty.clone(), val.clone()))
+                }
+            })
+            .map(|val| val.as_ref())
+    }
+
+    fn len(&self) -> usize {
+        self.values.len()
+    }
+}
+
+// Migration SQL generation (reuse SQLite logic since Turso is compatible)
+fn sql_for_op(current: &mut ADB, op: &Operation) -> Result<String> {
+    match op {
+        Operation::AddTable(tbl) => Ok(create_table(tbl, false)),
+        Operation::AddTableIfNotExists(tbl) => Ok(create_table(tbl, true)),
+        Operation::RemoveTable(name) => Ok(drop_table(name)),
+        Operation::AddColumn(tname, col) => add_column(tname, col),
+        Operation::RemoveColumn(tname, colname) => remove_column(current, tname, colname),
+        Operation::ChangeColumn(tname, old, new) => {
+            Ok(change_column(current, tname, old, Some(new)))
+        }
+        Operation::AddTableConstraints(_) | Operation::RemoveTableConstraints(_) => {
+            // Turso/SQLite doesn't support adding/removing constraints separately
+            // They must be included when creating the table
+            Ok(String::new())
+        }
+    }
+}
+
+fn create_table(table: &ATable, if_not_exists: bool) -> String {
+    let mut constraints: Vec<String> = Vec::new();
+    let mut defs: Vec<String> = table.columns.iter().map(define_column).collect();
+    for column in &table.columns {
+        if column.reference().is_some() {
+            constraints.push(define_constraint(column));
+        }
+    }
+    defs.append(&mut constraints);
+
+    let table_name = helper::quote_reserved_word(&table.name);
+    let prefix = if if_not_exists {
+        format!("CREATE TABLE IF NOT EXISTS {table_name} (")
+    } else {
+        format!("CREATE TABLE {table_name} (")
+    };
+
+    // Format with newlines if it would be longer than 120 characters
+    let single_line = format!("{}{});", prefix, defs.join(", "));
+    if single_line.len() <= 120 {
```

Comment:
just noting hard coded 120 line length in here.
a more generic approach needed, but beyond scope of this PR